### PR TITLE
ha-robustness failure: "The lock session requested to start is already in use"

### PR DESCRIPTION
**Pull request by MishaDemianenko**
Update dead client and idle channel reaper to cleanup clients locks session to prevent failure on next connection from the same client with same RequestContext.

Cleanup.

### CODE REVIEW

**Comment by tinwelint** [comment]
@MishaDemianenko needs rebase

File: community/kernel/src/main/java/org/neo4j/kernel/impl/util/collection/TimedRepository.java [hunk 1]
```diff
@@ -117,6 +117,16 @@ public void begin( KEY key ) throws ConcurrentAccessException
         }
     }
 
+    public void removeLockSession( KEY key )
```
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/util/collection/TimedRepository.java]: Why is this method called `removeLockSession`? `TimedRepository` is a generic class. Also check code style.

**Comment by MishaDemianenko** [comment]
@tinwelint @digitalstain new version with cleaned `Master` and start of `MasterImpl` cleanup.

**Comment by tinwelint** [comment]
I like the new changes, I'll merge on green build.

**Pull request merged**